notebooks/eda-text.ipynb

```diff
@@ -0,0 +1,509 @@
+      "text/plain": [
+       "<Figure size 1440x720 with 1 Axes>"
+      ]
+     },
+     "metadata": {
+      "needs_background": "light"
+     },
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "data[\"title_length_char\"] = data[\"title\"].str.len()\n",
+    "p_value = normaltest(data[\"title_length_char\"])[1]\n",
+    "plt.figure(figsize=(20, 10))\n",
+    "plt.title(\n",
+    "    f\"Distribution of title length in characters \\\n",
+    "    (Normal test p-value:{p_value} Accept:{p_value < CONFIDENCE})\"\n",
+    ")\n",
+    "sns.histplot(data=data, x=\"title_length_char\",\n",
+    "             kde=True, stat=\"density\", bins=30)\n",
+    "plt.xlabel(\"Length\")\n",
+    "plt.show()\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Text cleaning"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "data[\"title\"] = data[\"title\"].str.lower()\n",
+    "data[\"title\"] = data[\"title\"].str.replace(r\"[^\\w\\s]\", \"\", regex=True)\n",
+    "data[\"title\"] = data[\"title\"].str.replace(r\"\\s+$\", \"\", regex=True)\n",
+    "data[\"title\"] = data[\"title\"].str.replace(r\"x\\d+$\", \"\", regex=True)\n",
+    "data[\"title\"] = data[\"title\"].str.replace(r\"\\W*\\b\\w{1,3}\\b\", \"\", regex=True)\n"
```

Comment:
Maybe, it would be better to create function for text cleaning and use pandas `apply` method. I think, you should also try other techniques. Keeping cleaned data in separate dataframe column is recommended

Reply:
Vectorized pandas functions are much faster than user defined applied to each record